client/components/domains/register-domain-step/index.jsx

```diff
@@ -334,24 +363,57 @@ class RegisterDomainStep extends React.Component {
 		this.props.onSave( this.state );
 	};
 
-	onSearchChange = searchQuery => {
+	repeatSearch = () => {
+		this.onSearchChange( this.state.lastQuery, () => this.onSearch( this.state.lastQuery ) );
+	};
+
+	getSetFiltersForAPI() {
+		const { filters } = this.state;
+		return {
+			...mapKeys(
+				pickBy( filters, value => isNumberString( value ) || typeof value === 'boolean' ),
+				( value, key ) => snakeCase( key )
+			),
+		};
+	}
+
+	onFiltersChange = newFilters => {
+		this.setState( {
+			filters: newFilters,
+		} );
+	};
+
+	onFiltersReset = () => {
```

Comment:
From what I understood from the design, I think the reset button is supposed to clear all active filters across all inputs. @danhauk, can you confirm?

Reply:
@jsnmoon I think it would provide a better user experience if the reset button just cleared the active set of filters as @hambai mentioned. Clearing everything might cause some confusion as it's not immediately clear it would reset even the non-active filters.